src/cuda/list_sampling.cu

```diff
@@ -0,0 +1,88 @@
+#include <curand_kernel.h>
+#include <tuple>
+#include "atomic.h"
+#include "cuda_common.h"
+#include "tensor_ops.h"
+
+namespace gs {
+namespace impl {
+
+/**
+ * @brief ListSampling, using A-Res sampling for replace = False and uniform
+ * sampling for replace = True. It will return (selected_data, selected_index)
+ *
+ * @tparam IdType
+ * @param data
+ * @param num_picks
+ * @param replace
+ * @return std::tuple<torch::Tensor, torch::Tensor>
+ */
+template <typename IdType>
+std::tuple<torch::Tensor, torch::Tensor> _ListSampling(torch::Tensor data,
```

Comment:
As suggested by Gan Quan in the quip document, can we implement `ListSampling` by `torch.multimonial`?

Reply:
However, `torch.multimonial` cannot support over 2^24 (16,777,216) items sampling on GPU.